### PR TITLE
Fix no scroll on touch devices

### PR DESCRIPTION
**Pull request by souporserious**
Fixed this a while ago on an [old modal library](https://github.com/souporserious/expose-modal) I made. Thought I could help here :) Here is a [Codepen](http://s.codepen.io/souporserious/debug/3bfdb4469745002b8a68894d4328703e) if you want an easy way to test it on a touch device.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,39 +1,44 @@
 (function(root) {
+  var docEl = document && document.documentElement;
   var scrollbarSize;
+  var scrollTop;
 
   function getScrollbarSize() {
     if (typeof scrollbarSize !== 'undefined') return scrollbarSize;
 
-    var doc = document.documentElement;
```
Comment (davidtheclark) [inline index.js:7]: These invocations of `document` were deliberately placed inside the functions in order to ensure compatibility with server-rendered React apps.

File: index.js [hunk 2]
```diff
@@ -1,39 +1,44 @@
...
-    if (typeof document === 'undefined') return;
-    var doc = document.documentElement;
-    var rightPad = parseInt(getComputedStyle(doc)['padding-right'], 10);
-    doc.style.overflow = 'hidden';
+    if (!docEl) return;
+    scrollTop = window.pageYOffset;
     if (hasScrollbar()) {
-      rightPad += getScrollbarSize();
-      doc.style.paddingRight = rightPad + 'px';
+      docEl.style.width = 'calc(100% - '+ getScrollbarSize() +'px)';
```
Comment (davidtheclark) [inline index.js]: Why is this change necessary?

**Comment by davidtheclark** [comment]
@souporserious The new trick seems to work great in all the browsers I can access right now — so cool! As noted in comments, there are a few code changes that seem unrelated to the primary fix that I'm not sure about.

I'm thinking maybe this should be a breaking change release, since it fundamentally changes one of the two simple things this module does and I can't be 100% sure it's not going to cause a regression for somebody. Does that make sense?

File: index.js (hunk 1 above)
Comment (souporserious) [inline index.js:7]: Isn't caching it above ok though? Since it will still come back as false if it was rendered server side. Then we don't have to grab it every time.

File: index.js (hunk 2 above)
Comment (souporserious) [inline index.js]: It's not necessary, but it gets rid of having to run `getComputedStyle` and still provides the same outcome and browser support as before.

**Comment by souporserious** [comment]
Awesome! Commented back on those. I can revert it back if you need me to, just thought I could do some small optimizations while I was in there. Yeah that makes sense as a breaking change. I can update the CHANGELOG as well. Forgot about that.

File: index.js (hunk 1 above)
Comment (davidtheclark) [inline index.js:7]: You could _cache_ it above but the current line 2 actually looks for it as soon as the module is parsed. Besides, I think it's just a change better left to another PR.

File: index.js (hunk 2 above)
Comment (davidtheclark) [inline index.js]: Yeah, browser support will be the same. I'm fine with it. In the future it's a good idea to separate completely independent ideas like this into different PRs/commits.

**Comment by souporserious** [comment]
@davidtheclark sorry about that. Cleaned it up. I just force pushed over so it wasn't a bunch of weird commits.

**Comment by souporserious** [comment]
@davidtheclark just pinging you on this. Please let me know if you need anything else from me.

File: index.js [hunk 3]
```diff
@@ -1,50 +1,55 @@
 (function(root) {
-  var scrollbarSize;
+    var scrollbarSize;
+    var scrollTop;
```
Comment (davidtheclark) [inline index.js]: Why have you reindented everything?

**Comment by davidtheclark** [comment]
Sorry, @souporserious — I'm busier than I'm used to, so slowing down a little on side projects. Still want to merge this, but it looks like the indentation is changed now?

**Comment by souporserious** [comment]
@davidtheclark sorry about that. Not sure what happened. Fixed now 👍

**Comment by davidtheclark** [comment]
Cool, probably your editor did it or something. I will try this out again tonight then probably merge & release.

**Comment by souporserious** [comment]
Awesome, thank you!

**Comment by davidtheclark** [comment]
@souporserious When I pulled this branch down and tried it out, it worked on my desktop browsers but _not_ on my mobile safari: that is, I was still able to scroll on mobile safari. Are you having a different experience?

(In `master` I set up a little script to run browser-sync that helps with this testing. May want to rebase this branch onto master to use that.)

**Comment by souporserious** [comment]
@davidtheclark hmm I rebased and pushed. I'm unable to get it to scroll on any of my devices. I tested with my Browserstack account and on my personal iPhone 6s and was unable to get it to scroll when activated. Any specific version/sandbox you're testing on?

**Comment by davidtheclark** [comment]
@souporserious I must have bundled incorrectly or something. Tried again and it worked great. Will cut a new release now.

**Comment by davidtheclark** [comment]
Published in v2. Thanks again!

**Comment by souporserious** [comment]
Thank you! Glad I was able to share that knowledge :)
